packages/opentelemetry-web/src/WebTracer.ts

```diff
@@ -14,25 +14,43 @@
  * limitations under the License.
  */
 
+import { BasePlugin } from '@opentelemetry/core';
 import { BasicTracer, BasicTracerConfig } from '@opentelemetry/tracing';
 import { StackScopeManager } from './StackScopeManager';
 
+/**
+ * WebTracerConfig provides an interface for configuring a Web Tracer.
+ */
+export interface WebTracerConfig extends BasicTracerConfig {
+  /**
+   * plugins to be used with tracer, they will be enabled automatically
+   */
+  plugins?: BasePlugin<unknown>[];
+}
+
 /**
  * This class represents a web tracer with {@link StackScopeManager}
  */
 export class WebTracer extends BasicTracer {
   /**
    * Constructs a new Tracer instance.
+   * @param {WebTracerConfig} config Web Tracer config
    */
-  /**
-   *
-   * @param {BasicTracerConfig} config Web Tracer config
-   */
-  constructor(config: BasicTracerConfig = {}) {
+  constructor(config: WebTracerConfig = {}) {
     if (typeof config.scopeManager === 'undefined') {
       config.scopeManager = new StackScopeManager();
     }
-    config.scopeManager.enable();
+    if (typeof config.plugins === 'undefined') {
+      config.plugins = [];
+    }
     super(Object.assign({}, { scopeManager: config.scopeManager }, config));
+
+    // enable scope manager
+    config.scopeManager.enable();
+
+    // enable all plugins
+    for (let i = 0, j = config.plugins.length; i < j; i++) {
```

Comment:
Could you use `for (const .. of` here?

The TS compiler can have various output settings to make sure that older browser could get appropriate polyfills for it, and so I think we should be coding with the latest spec